File: Hackerank/Days 10/main.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');
    main();
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    const n = parseInt(readLine().trim(), 10);
    
    const binary = n.toString(2);
    const onesGroups = binary.split('0');
    let maxOnes = 0;

    for (let group of onesGroups) {
        if (group.length > maxOnes) {
            maxOnes = group.length;
        }
    }

    console.log(maxOnes);
}

File: Hackerank/Days 10/main.cs
using System;

class Solution {
    static void Main(string[] args) {
        int n = Convert.ToInt32(Console.ReadLine().Trim());
        
        string binary = Convert.ToString(n, 2);
        int maxOnes = 0;
        int currentOnes = 0;

        foreach (char c in binary) {
            if (c == '1') {
                currentOnes++;
                if (currentOnes > maxOnes) {
                    maxOnes = currentOnes;
                }
            } else {
                currentOnes = 0;
            }
        }

        Console.WriteLine(maxOnes);
    }
}
